tango/step_cache.py

```diff
@@ -1,11 +1,13 @@
 import logging
 from abc import abstractmethod
 from dataclasses import dataclass
-from typing import Any, TypeVar
+from typing import Any, TypeVar, Union
 
-from tango.common.from_params import FromParams
-from tango.common.registrable import Registrable
-from tango.step import Step
+from .common.from_params import FromParams
+from .common.registrable import Registrable
+from .format import Format
+from .step import Step
+from .step_info import StepInfo
```

Comment:
I thought we don't like relative imports? Don't they cause some sort of trouble with `Registrable`?

Reply:
I don't not like (explicit) relative imports. I prefer them actually, and I'm not aware of any issues with `Registrable`.

Reply:
I also prefer relative imports, but working on the AllenNLP codebase educated me out of it. I'm fine, happy even, to try again in Tango.